src/nrnoc/secref.cpp

```diff
@@ -332,6 +332,12 @@ Section* nrn_sectionref_steer(Section* sec, Symbol* sym, int* pnindex) {
                 hoc_execerror("SectionRef.child[index]", (char*) 0);
             }
         }
+        // modeldb 114355 uses legacy syntax SectionRef.child(i). Allow
+        // though there is no ndim on stack.
+        bool ok = hoc_stack_type_is_ndim(0) ? (hoc_pop_ndim() == 1) : (*pnindex == 1);
+        if (!ok) {
+            hoc_execerror("SectionRef.child[index] has only one dimension", NULL);
```

Comment:
Just trying to parse the definition of `ok`...isn't the error triggered when the number of dimensions is **not** one?